### PR TITLE
Add a micrometer-backed Graphite reporter

### PR DESCRIPTION
**Pull request by ThaerAlAjlouni**

### CODE REVIEW

File: components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteConfig.java [hunk 1]
```diff
@@ -30,16 +30,19 @@
...
     private final String prefix;
+    private final boolean enabled;
 
     @JsonCreator
     GraphiteConfig(@JsonProperty("host") String host,
                    @JsonProperty("port") Integer port,
                    @JsonProperty("intervalMillis") Long intervalMillis,
-                   @JsonProperty("prefix") String prefix) {
+                   @JsonProperty("prefix") String prefix,
+                   @JsonProperty("enabled") String enabled) {
```
Comment (Noddy76) [inline components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteConfig.java:40]: This should probably be a `boolean` or a `Boolean` to be null safe. Either way it should be the same type as the getter.

File: components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteConfig.java [hunk 2]
```diff
@@ -62,16 +65,23 @@ public long intervalMillis() {
...
 
+    @JsonProperty("enabled")
+    public boolean enabled() {
+        return enabled;
+    }
+
     @Override
     public String toString() {
-        return new StringBuilder(96)
+        return new StringBuilder()
```
Comment (Noddy76) [inline components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteConfig.java:75]: NBD: I realise that this is just how you found it but there is no advantage over string concatenation to using `StringBuilder`.

File: components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteConfig.java (hunk 1 above)
Comment (ThaerAlAjlouni) [reply components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteConfig.java:40]: What led me to use a `String` is to default the value to `true` when the `enabled` property is missing (to maintain backward compatibility) while using a `boolean` will always default to `false` with `@JsonCreator` if the property is missing.

File: components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteConfig.java (hunk 2 above)
Comment (ThaerAlAjlouni) [reply components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteConfig.java:75]: I agree, will simplify this implementation.

**Pull request merged**